platforms/core-runtime/process-services/src/main/java/org/gradle/process/internal/ExecHandleBuilder.java

```diff
@@ -20,6 +20,12 @@
 import org.gradle.process.ExecSpec;
 import org.gradle.process.ProcessForkOptions;
 
+/**
+ * Deprecated. Use {@link ClientExecHandleBuilder} instead. Kept for now since it's used by the Kotlin plugin.
+ *
+ * Can be merged with {@link ClientExecHandleBuilder} in Gradle 9.0.
+ */
+@Deprecated
 public interface ExecHandleBuilder extends ProcessForkOptions, BaseExecSpec, ExecSpec {
```

Comment:
💭 This interface was not changed for binary compatilibility reasons: since Kotlin uses it